### PR TITLE
Add rpm spec file

### PR DESCRIPTION
**Pull request by champtar**
This adds a spec file so we can build rpm files for Fedora and CentOS
https://copr.fedorainfracloud.org/coprs/champtar/dracut-sshd/

If you want to manually build it you need to use "rpkg"

### CODE REVIEW

File: 46sshd/module-setup.sh [hunk 1]
```diff
@@ -65,6 +65,10 @@ install() {
     # include those files
     inst_multiple -o /etc/systemd/network/*
 
+    # Add command to unlock luks volumes to bash history for easier use
+    echo "systemd-tty-ask-password-agent" >> "$initdir/root/.bash_history"
```
Comment (gsauthof) [inline 46sshd/module-setup.sh]: Nice idea! This even works when bash is invoked as /bin/sh (as it is in an early userspace ssh session).

File: 46sshd/sshd_config [hunk 2]
```diff
@@ -7,6 +7,7 @@ GSSAPIAuthentication no
 GSSAPICleanupCredentials no
 UsePAM no
 X11Forwarding no
+AuthenticationMethods publickey
```
Comment (gsauthof) [inline 46sshd/sshd_config]: Makes sense.

File: dracut-sshd.spec [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+%prep
+{{{ git_dir_setup_macro }}}
+
+%build
+# nothing to do here
+
+%install
+rm -rf %{buildroot}
+mkdir -p %{buildroot}/usr/lib/dracut/modules.d/
+mv 46sshd %{buildroot}/usr/lib/dracut/modules.d/
```
Comment (gsauthof) [inline dracut-sshd.spec]: Can you change that to `cp -r ...`?

Such that your source tree isn't modified.

File: dracut-sshd.spec [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+Add sshd to your initramfs
+
+%prep
+{{{ git_dir_setup_macro }}}
+
+%build
+# nothing to do here
+
+%install
+rm -rf %{buildroot}
```
Comment (gsauthof) [inline dracut-sshd.spec]: Do we need this rm?

File: dracut-sshd.spec [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+mkdir -p %{buildroot}/usr/lib/dracut/modules.d/
+mv 46sshd %{buildroot}/usr/lib/dracut/modules.d/
+
+%clean
+rm -rf %{buildroot}
+
+%files
+%attr(755, root, root) /usr/lib/dracut/modules.d/46sshd/module-setup.sh
+%attr(644, root, root) /usr/lib/dracut/modules.d/46sshd/sshd.service
+%attr(644, root, root) %config(noreplace) /usr/lib/dracut/modules.d/46sshd/sshd_config
```
Comment (gsauthof) [inline dracut-sshd.spec]: Do we meed these explicit %attr declarations?

I mean the permissions are already fine in the git tree and thus they shouldn't change when rpm copies them.

Do you mind also installing the readme? For example:

    %doc README.md

File: dracut-sshd.spec [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+%changelog
+
```
Comment (gsauthof) [inline dracut-sshd.spec]: Do you mind adding a simple changelog entry?

File: 46sshd/sshd_config (hunk 2 above)
Comment (gsauthof) [reply 46sshd/sshd_config]: I've integrated this in https://github.com/gsauthof/dracut-sshd/commit/12a0d4105c171179edae323d7cc9cd65643f9990

File: 46sshd/module-setup.sh (hunk 1 above)
Comment (gsauthof) [reply 46sshd/module-setup.sh]: Cherry-picked this with  https://github.com/gsauthof/dracut-sshd/commit/853872061f372208b26230359fa7ee7717ddd9a5

**Comment by gsauthof** [comment]
Ok, so I've added a .spec to the repository (with the changes I described in the code review comments and more) and created a Copr repository:

https://copr.fedorainfracloud.org/coprs/gsauthof/dracut-sshd/

**Comment by champtar** [comment]
Hi @gsauthof,

Thanks for taking care of everything !!

**Pull request closed without merging**